go-controller/pkg/ovn/policy.go

```diff
@@ -185,24 +191,40 @@ func hashedPortGroup(s string) string {
 	return util.HashForOVN(s)
 }
 
+func generatePrefix(s string) sets.String {
+	prefixes := sets.NewString()
+	for index := range s {
+		prefixes.Insert(s[:index])
+	}
+	prefixes.Insert(s)
+	return prefixes
+}
+
 // updateStaleDefaultDenyACLNames updates the naming of the default ingress and egress deny ACLs per namespace
 // oldName: <namespace>_<policyname> (lucky winner will be first policy created in the namespace)
 // newName: <namespace>_egressDefaultDeny OR <namespace>_ingressDefaultDeny
 func (oc *Controller) updateStaleDefaultDenyACLNames(npType knet.PolicyType, gressSuffix string) error {
 	cleanUpDefaultDeny := make(map[string][]*nbdb.ACL)
+	prefixes := generatePrefix(gressSuffix)
 	p := func(item *nbdb.ACL) bool {
+		aclNameSuffix := strings.Split(*item.Name, "_")
+		if len(aclNameSuffix) == 1 {
+			// doesn't have suffix; no update required; append the actual suffix since this ACL can be skipped
+			aclNameSuffix = append(aclNameSuffix, gressSuffix)
+		}
 		return item.ExternalIDs[defaultDenyPolicyTypeACLExtIdKey] == string(npType) && // default-deny-policy-type:Egress or default-deny-policy-type:Ingress
 			strings.Contains(item.Match, gressSuffix) && // Match:inport ==	@ablah80448_egressDefaultDeny or Match:inport == @ablah80448_ingressDefaultDeny
-			!strings.Contains(*item.Name, arpAllowPolicySuffix) && // != name: namespace_ARPallowPolicy
-			!strings.Contains(*item.Name, gressSuffix) // filter out already converted ACLs (we still continue to pick the cases where names were >45 chars)
+			!strings.Contains(item.Match, staleArpAllowPolicyMatch) && // != match: (arp || nd) OR arp
```

Comment:
The `staleArpAllowPolicyMatch` ACLs are deleted later on the sync, so it seems that we would not have to account for them here if we moved `updateStaleDefaultDenyACLNames` after that section. 
In general, in the sync section, we should delete ACLs first so that they are not considered in later predicates.
But this is how it was before, so up to you.

Reply:
ack! this is something I can change. I am adding some more corner case tests into the unit test, let me push this and that together.